daemon/stats_collector.go

```diff
@@ -76,22 +76,43 @@ func (s *statsCollector) unsubscribe(c *Container, ch chan interface{}) {
 }
 
 func (s *statsCollector) run() {
+	type publishersPair struct {
+		container *Container
+		publisher *pubsub.Publisher
+	}
+	// cannot determine capacity of an array here,
+	// it grows enough after the first iteration
+	pairs := make([]publishersPair, 0)
```

Comment:
If create it inside the loop I  will have no chance to reuse memory. The slice would be recreated each time.
Slice of slice reuse the same buffer.
http://play.golang.org/p/58Uvg3xYIB and https://code.google.com/p/go-wiki/wiki/SliceTricks#Filtering_without_allocating
Of course, in the first iteration it has not enoguh capacity to store all pairs, but it wil save its capacity for the next iteration.